GenderPayGap.WebUI/Search/CachedObjects/SearchCachedOrganisation.cs

```diff
@@ -16,16 +20,75 @@ public class SearchCachedOrganisation
         public string EmployerReference { get; set; }
         public OrganisationStatuses Status { get; set; }
         public int MinEmployees { get; set; }
-        public List<OrganisationSizes> OrganisationSizes { get; set; }
         public List<int> ReportingYears { get; set; }
-        public DateTime DateOfLatestReport { get; set; }
-        public bool ReportedWithCompanyLinkToGpgInfo { get; set; }
-        public bool ReportedLate { get; set; }
         public List<string> SicCodeIds { get; set; }
         public List<char> SicSectionIds { get; set; }
         public List<SearchReadyValue> SicCodeSynonyms { get; set; }
         public bool IncludeInViewingService { get; set; }
         public SectorTypes Sector { get; set; }
+        public Dictionary<int, List<OrganisationSizes>> ReportingYearToOrganisationSizesMap { get; set; }
+        public List<int> ReportedLateYears { get; set; }
+        public List<int> ReportedWithCompanyLinkToGpgInfoYears { get; set; }
+        public Dictionary<int, DateTime> ReportingYearToDateOfLatestReportMap { get; set; }
+        
+        public List<OrganisationSizes> OrganisationSizes(List<int> years)
+        {
+            if (years.IsNullOrEmpty())
+            {
+                years = ReportingYearsHelper.GetReportingYears();
+            }
 
+            var organisationSizes = new List<OrganisationSizes>();
+
+            foreach (var year in years)
+            {
+                if (ReportingYearToOrganisationSizesMap.TryGetValue(year, out List<OrganisationSizes> sizes))
+                {
+                    organisationSizes = organisationSizes.Union(sizes).ToList();
+                }
+            }
+
+            return organisationSizes;
+        }
+
+        public List<DateTime> DatesOfLatestReports(List<int> years)
+        {
+            if (years.IsNullOrEmpty())
+            {
+                years = ReportingYearsHelper.GetReportingYears();
+            }
+
+            var latestReports = new List<DateTime>();
+
+            foreach (var year in years)
+            {
+                if (ReportingYearToDateOfLatestReportMap.TryGetValue(year, out DateTime latestDate))
+                {
+                    latestReports.Add(latestDate);
+                }
+            }
+
+            return latestReports;
+        }
+
+        public bool ReportedWithCompanyLinkToGpgInfo(List<int> years)

```

Comment:
I find this name quite confusing and same for `ReportedWithCompanyLinkToGpgInfoYears`, how about `HasReportedWithCompanyLinkInAnyYear` or something?